identity/context.go

```diff
@@ -0,0 +1,26 @@
+package identity
+
+import (
+	"context"
+
+	"github.com/MichaelMure/git-bug/repository"
+)
+
+// identityCtxKey is a unique context key, accessible only in this struct.
+type identityCtxKey struct {
+	repo string
+}
+
+// AttachToContext attaches an Identity to a context.
+func AttachToContext(ctx context.Context, r repository.RepoCommon, u *Identity) context.Context {
+	return context.WithValue(ctx, identityCtxKey{r.GetPath()}, u)
```

Comment:
After more thought spent, I think you should store the identity's ID instead of the full identity:
- the repo to search in is already known from the request: the GraphQL query has it already. The git file upload and retrieval handler don't, but that's because they only work with a single repo for now. This will need to change.
- a search in the cache with and identity ID is instant.
- it will makes things easier in the mutation resolver: instead of bypassing the cache as you are doing now, you just need to load the identity and everything is done for you

Reply:
The reason I added the identity ID was to (in a multi-repo world) avoid people from e.g. adding bugs to a repo in which their identity doesn't exist, but I don't know enough about what your plan for multi-repo support is.